src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -622,12 +622,15 @@ fun filterDeriveProcMacros(processor: RsResolveProcessor): RsResolveProcessor =
         true
     }
 
-fun RsOuterAttributeOwner.shouldHideElementInCompletion(contextMod: RsMod): Boolean {
+fun RsOuterAttributeOwner.shouldHideElementInCompletion(context: RsElement, contextMod: RsMod): Boolean {
     val elementContainingMod = containingMod
     if (elementContainingMod == contextMod) return false
 
     // Hide `#[doc(hidden)]` items
-    if (queryAttributes.isDocHidden) return true
+    val isDeriveMacroInImport = context.ancestorStrict<RsUseItem>() != null
+        && this is RsFunction && isCustomDeriveProcMacroDef
+        && containingCrate != contextMod.containingCrate
+    if (queryAttributes.isDocHidden && !isDeriveMacroInImport) return true
```

Comment:
Let's calculate the second condition more lazy:
```suggestion
    if (queryAttributes.isDocHidden) {
        val isDeriveMacroInImport = context.ancestorStrict<RsUseItem>() != null
            && this is RsFunction && isCustomDeriveProcMacroDef
            && containingCrate != contextMod.containingCrate
        if (!isDeriveMacroInImport) return true
    }
```